File: core-query/src/main/kotlin/ru/patterns/corequery/domain/Transaction.kt
package ru.patterns.corequery.domain

import java.time.LocalDateTime

/**
 * Транзакция
 */
data class Transaction(
    /**
     * Идентификатор транзакции
     */
    val id: TransactionId,
    /**
     * Время совершения транзакции
     */
    val transactionTimestamp: LocalDateTime,
    /**
     * Идентификатор счета, откуда списываются средства,
     * может быть null, в случае, если происходит пополнение счета
     */
    val accountFrom: AccountId?,
    /**
     * Идентификатор счета, куда переводятся средства
     */
    val accountTo: AccountId,
    /**
     * Сумма, которую требуется перевести
     */
    val amount: Balance,
    /**
     * Владелец счета, с которого произведен перевод
     */
    val clientId: ClientId,
)

File: client-interface/src/main/resources/static/client/js/models/Transaction.js
import {BaseModel} from './BaseModel.js';

export class Transaction extends BaseModel {

    constructor(data = {}) {
        super(data);
        this.id = data.id || '';
        this.amount = data.amount || 0;
        this.accountFrom = data.accountFrom || '';
        this.accountTo = data.accountTo || '';
        this.transactionTimestamp = data.transactionTimestamp || '';
    }

    getType(accountId) {
        return this.accountFrom === accountId ? 'Списание' : 'Пополнение';
    }

    getFormattedAmount(accountId) {
        const prefix = this.accountFrom === accountId ? '-' : '+';
        return `${prefix} ${this.amount} ₽`;
    }
}
